arkouda/timeclass.py

```diff
@@ -144,6 +143,158 @@ def __init__(self, array, unit: str = _BASE_UNIT):  # type: ignore
         # Deprecated
         self._data = self.values
 
+        if len(self.values):
+            # Year, hour, minute, second
+            seconds = self.values // _get_factor("s")
+            days = cast((seconds // 86400), int64)
+            self._year = cast((days // 365.24 + 1970), int64)
+            self._month = cast(((seconds % 31556926) / 2629743), int64) + 1
+            self._day = cast((((seconds % 31556926) % 2629743) / 86400), int64) + 2
+
+            self._hour = cast((self.values / _get_factor("h")), int64)
+            self._hour %= 24
+            self._minute = cast((self.values / _get_factor("m")), int64)
+            self._minute %= 60
+            self._second = cast((self.values / _get_factor("s")), int64)
+            self._second %= 60
+
+            # Month, day
+            self._day = self.values // _get_factor("d")
+            self._day += 1401  # move starting point to 1996 - 03 - 01
+            self._day %= 1461  # max 3 years 365 day (not 4 years!) remaining
+            d = self._day[:]
+            self._day[d >= 731] -= 731
+
+            d = self._day[:]
+            self._day[d >= 365] -= 365
+
+            d = self._day[:]
+            self._month = d[:] - d[:] + 3
+            self._day[d >= 306] -= 306
+            self._month[d >= 306] = 1
+
+            d = self._day[:]
+            self._day[d >= 153] -= 153
+            self._month[d >= 153] = 8
+
+            d = self._day[:]
+            self._day[d >= 122] -= 122
+            self._month[d >= 122] += 4
+            d = self._day[:]
+            self._day[d >= 61] -= 61
+            self._month[d >= 61] += 2
```

Comment:
or we could store the boolean pdarray to avoid recomputation since we use it more than once
```python
self._month = self._day - self._day + 3
ge_306 = self._day >= 306
self._month[ge_306] = 1
self._day[ge_306] -= 306
```